docs/source/bufr4_descriptors.rst

```diff
@@ -0,0 +1,172 @@
+
+.. role:: redtext
+
+
+Introduction
+------------
+
+As part of the BUFR format, a list of the included parameters is embedded within the file using the
+:redtext:`unexpandedDescriptors` element of Section 3.
+This element takes a list of the parameters to be included and the order of those parameters.
+For example, the unexpanded descriptor list (in text, see format of descriptors below):
+
+.. code::
+
+	["unexpandedDescriptors"] = ["wigosID", "year", "month", "day", "hour", "minute", \
+                                 "latitude", "longitude", "pressure reduced to mean sea level"]
+
+would specify that the station identifier (wigosID) followed by the date, time, location and then the pressure reduced to mean sea level would be included in the data section.
+For conciseness, aliases (or sequences in BUFR terminology) exist to group commonly
+reported parameters together, for example grouping the year, month and day together in the group date.
+Using sequences, the above example becomes:
+
+.. code::
+
+	["unexpandedDescriptors"] = ["wigosID", "date", "time", "location", "pressure reduced to mean sea level"]
+
+where
+
+.. code::
+
+	["date"] = ["year", "month", "day"]
+	["time"] = ["hour", "minute"]
+	["location"] = ["latitude", "longitude"]
+
+These sequences form building blocks that can be used to create much longer sequences using a small number of descriptors.
+
+Format of BUFR descriptors
+--------------------------
+
+Whilst text strings have been used above to represent the parameters to report this has been done for ease of reading
+and explanation.
+Within the BUFR format these are specified by 6 digit codes of the form FXXYYY, with each of F, XX and YYY having
+specific meaning.
+
+* F: Type of BUFR descriptor
+
+   - 0: element descriptor (BUFR Table B);
+   - 1: replication (or repetition) descriptor;
+   - 2: operator (BUFR Table C);
+   - 3: sequence descriptor (BUFR Table D)
+
+* XX: Sub-table (class) within type of descriptor
+* YYY: Index within in sub-table.
+
+As an example, the table below shows the first few entries from BUFR Table B 01.
+
+For the above example, using the FXXYYY notation, the list of abbreviated (or unexpanded) descriptors becomes:
+
+.. code::
+
+    ["unexpandedDescriptors"] = [301150, 301011, 301012, 301023, 010051]
+
+where:
+
+.. code::
+
+    [301150] = [001125, 001126, 001127, 001128] # (wigosID)
+             = ["WIGOS identifer series", "WIGOS identifier issuer", "WIGOS identifier issue number", "WIGOS local identifier"]
+    [301011] = [004001,  004002, 004003] # (date)
+             = ["year", "month",  "day"]
+    [301012] = [004004,   004005] # (time)
+             = ["hour", "minute"]
+    [301023] = [    005002,      006002] # (location)
+             = ["latitude", "longitude"]
+    [010051] = ["pressure reduced to mean sea level"]
+
+
+Replication / repetition
+------------------------
+
+Within the BUFR format elements can be repeated using the replication descriptors (F=1 in FXXYYY).
```

Comment:
Within the BUFR format, elements can be ...

Reply:
done